rust_snuba/src/utils/metrics/backends/datadog.rs

```diff
@@ -0,0 +1,136 @@
+use rust_arroyo::utils::metrics::{MetricsClientTrait};
+use dogstatsd;
+
+use super::abstract_backend::MetricsBackend;
+pub struct DatadogMetricsBackend {
+    client_sd: dogstatsd::Client,
+    _tags: Vec<String>,
+}
+
+impl MetricsBackend for DatadogMetricsBackend {
+    fn events(
+        &self,
+        title: &str,
+        text: &str,
+        _alert_type: &str,
+        _priority: &str,
+        tags: &[&str],
+    ) {
+        // TODO figure out how to send priority and alert_type
+        self.client_sd.event(title, text, tags ).unwrap()
+    }
+}
+
+
+impl DatadogMetricsBackend {
+    pub fn new(host: String, port: u16 , tags:Vec<String>) -> Self {
+
+        let host_port: String = format!("{host}:{port}");
+        let options = dogstatsd::OptionsBuilder::new()
+            .to_addr(host_port).build();
+
+        let client = dogstatsd::Client::new(options).unwrap();
+        DatadogMetricsBackend {
+            client_sd: client,
+            _tags: tags
+        }
+    }
+}
+
+impl MetricsClientTrait for DatadogMetricsBackend {
+    fn counter(
+        &self,
+        key: &str,
+        value: Option<i64>,
+        tags: Option<std::collections::HashMap<&str, &str>>,
+        sample_rate: Option<f64>,
+    ) {
+        if !self.should_sample(sample_rate) {
+            return;
+        }
+
+        let tags_str: Vec<String> = tags.unwrap().iter().map(|(k, v)| format!("{k}:{v}")).collect();
+
+        match value {
+            Some(v) => {
+                for _ in 0..v.abs() {
+                    if v < 0 {
+                        self.client_sd.decr(key, &tags_str).unwrap();
+                    } else {
+                        self.client_sd.incr(key, &tags_str).unwrap();
+                    }
+                }
```

Comment:
should this logic be replaced by https://docs.rs/dogstatsd/latest/dogstatsd/struct.Client.html#method.count ?

Seems inefficient to send `n` decr messages vs. subtract `n` if that's possible

Reply:
Yeah that was so dumb. Updated.